flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerConfiguration.java

```diff
@@ -68,11 +74,24 @@ public static SlotManagerConfiguration fromConfiguration(Configuration configura
 				"value " + AkkaOptions.ASK_TIMEOUT + '.', e);
 		}
 
-		final Time slotRequestTimeout = Time.milliseconds(
-				configuration.getLong(ResourceManagerOptions.SLOT_REQUEST_TIMEOUT));
+		final Time slotRequestTimeout = getSlotRequestTimeout(configuration);
 		final Time taskManagerTimeout = Time.milliseconds(
 				configuration.getLong(ResourceManagerOptions.TASK_MANAGER_TIMEOUT));
 
 		return new SlotManagerConfiguration(rpcTimeout, slotRequestTimeout, taskManagerTimeout);
 	}
+
+	private static Time getSlotRequestTimeout(final Configuration configuration) {
+		final long legacySlotRequestTimeoutMs = configuration.getLong(ResourceManagerOptions.SLOT_REQUEST_TIMEOUT);
+		final long slotRequestTimeoutMs;
+		if (legacySlotRequestTimeoutMs == ResourceManagerOptions.SLOT_REQUEST_TIMEOUT.defaultValue()) {
```

Comment:
I think `configuration.contains(ResourceManagerOptions.SLOT_REQUEST_TIMEOUT)` would be more accurate because one could set this option accidentally to the default value.